evm/bridge-token-factory/contracts/BridgeTokenFactoryWormhole.sol

```diff
@@ -0,0 +1,51 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.24;
+
+import {BridgeTokenFactory} from "./BridgeTokenFactory.sol";
+import "./Borsh.sol";
+
+interface IWormhole {
+    function publishMessage(
+        uint32 nonce,
+        bytes memory payload,
+        uint8 consistencyLevel
+    ) external payable returns (uint64 sequence);
+}
+
+contract BridgeTokenFactoryWormhole is BridgeTokenFactory {
+    IWormhole private _wormhole;
+    uint8 private _consistencyLevel;
+    uint32 public wormholeNonce;
+
+    function initializeWormhole(
+        address _tokenImplementationAddress,
+        address _nearBridgeDerivedAddress,
+        address wormholeAddress,
+        uint8 consistencyLevel
+    ) external initializer {
+        initialize(_tokenImplementationAddress, _nearBridgeDerivedAddress);
+        _wormhole = IWormhole(wormholeAddress);
+        _consistencyLevel = consistencyLevel;
+    }
+
+    function depositExtension(BridgeDeposit memory bridgeDeposit) internal override {
+        _wormhole.publishMessage(
+            wormholeNonce,
+            abi.encode(bridgeDeposit.token, bridgeDeposit.amount, bridgeDeposit.feeRecipient, bridgeDeposit.nonce),
+            _consistencyLevel
+        );
+
+        wormholeNonce++;
+
+    }
+
+    function withdrawExtension(string memory token, uint128 amount, string memory recipient) internal override {
+        _wormhole.publishMessage(
+            wormholeNonce,
+            abi.encode(token, amount, recipient),
```

Comment:
https://github.com/Near-One/omni-bridge/blob/main/near/omni-types/src/prover_result.rs#L9

Reply:
https://github.com/Near-One/omni-bridge/blob/main/near/omni-types/src/lib.rs#L197

We should also add info about fee and sender

Reply:
I added sender but fee is not currently implemented for withdrawals (it's not exclusive to wormhole). Let's discuss and add in the future PRs

Reply:
This should be covered in this issue 
https://github.com/Near-One/omni-bridge/issues/24